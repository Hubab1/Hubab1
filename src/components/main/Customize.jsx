import React from 'react';

export default class TalkAboutOptions extends React.Component {
    render () {
        return (
            <div>
                <div>
                    {['option1', 'option2', 'option3'].map((el, i) => (
                        <li key={i}>{el}</li>
                    ))}
                </div>
            </div>
        );
    }
}