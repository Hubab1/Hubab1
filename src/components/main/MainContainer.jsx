import React, { Component } from 'react';

import Terms from './Terms';
import TalkAboutOptions from './TalkAboutOptions';


class MainContainer extends Component {
  constructor () {
      super();
      this.state = {};
  }
  
  componentDidMount() {
      // fetch renter profile
      const renter_profile = {
          completed_terms_and_conditions: false,
          lets_talk_about: [],
          roommates: [],
          pets: [],
          guarantor: {}
      };
      if (!renter_profile.completed_terms_and_conditions) {
          this.setState({Page: Terms});
      } else if (renter_profile.lets_talk_about.length == 0) {
          this.setState({Page: TalkAboutOptions});
      }
  }
  
  render() {
      if (!this.state.Page) return null;
      return (
        <this.state.Page>
        </this.state.Page>
      );
  }
};

export default MainContainer;
