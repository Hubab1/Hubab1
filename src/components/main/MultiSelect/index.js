import React from 'react';
import MultiSelectContext from './context';

export default class MultiSelect extends React.Component {
    
    onClick = (name) => {
        if (this.props.value.indexOf(name) === -1) {
            this.props.onChange([...this.props.value.slice(), name]);
        } else {
            this.props.onChange(this.props.value.slice().filter(val => val !== name));
        }
    }
    render () {
        return (
            <MultiSelectContext.Provider value={{_onClick: this.onClick, _value: this.props.value}}>
                {React.Children.map(this.props.children, (child, i) => {
                    return React.cloneElement(child, {
                        _selected: this.props.value.indexOf(child.props.name) > -1
                    })
                })}
            </MultiSelectContext.Provider>
        );
    }
}