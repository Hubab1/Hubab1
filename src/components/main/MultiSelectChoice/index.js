import React from 'react';
import Button from '@material-ui/core/Button';
import MultiSelectContext from './context';

import { contentContainer, label, prefix, root, selected, unselected } from './styles';

export default class MultiSelectChoice extends React.Component {
    static contextType = MultiSelectContext;
    
    render () {
        return (
            <Button
                textTransform="none"
                onClick={()=>this.context._onClick(this.props.name)}
                variant="outlined"
                color="primary"
                type="submit"
                fullWidth
                classes={{
                    root: this.props.selected ? selected : unselected,
                    label: contentContainer
                }}
            >
                    <div className={prefix}>{this.props.prefix}</div>
                    <div className={label}>{this.props.label}</div>
            </Button>
        );
    }
}