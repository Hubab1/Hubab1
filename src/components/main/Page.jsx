import React from 'react';
import { connect } from 'react-redux';
import Button from '@material-ui/core/Button';

import { Banner, page, Link, subPage } from 'assets/emotion/styles';
import { nextScreen, selectors } from 'reducers/nav/reducer';
import { community } from 'constants.js';

export class Page extends React.Component {
    changeScreen = () => {
        this.props.nextScreen();
    }

    render () {
        return (
            <div className={page}>
                <Banner>{community.name}</Banner>
                <div className={subPage}>
                    {this.props.children}
                    <Button onClick={this.changeScreen} variant="contained" color="primary" type="submit" fullWidth>
                        {this.props.buttonText}
                    </Button>
                    <br/>
                    <br/>
                    <Link>Go Back</Link>
                </div>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    buttonText: selectors.getCurrentScreen(state).buttonText
});

export default connect(mapStateToProps, {nextScreen})(Page);