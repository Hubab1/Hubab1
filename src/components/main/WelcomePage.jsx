import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';

import { Bold, PageHeader, agentBlock, agentTitle, BackgroundImage } from 'assets/emotion/styles';
import history from 'history.js';

export class WelcomePage extends Component {

    render() {
        return (
            <Fragment>
                <div className={agentBlock}>
                    <BackgroundImage url={this.props.leaseSettings.background_image} />
                    <div>Jane Morgan</div>
                    <span aria-label="agent-avatar" role="img" style={{fontSize: 60, lineHeight: '80px'}}>ðŸ‘©</span>
                    <div className={agentTitle}>
                        <div>Leasing Agent</div>
                        <div>Sky Residencies</div>
                    </div>
                </div>
                <div style={{textAlign: 'left', padding: 25}}>
                    <PageHeader>Hi Sam,</PageHeader>
                    <div>Your magically simple rental application starts here.</div>
                    <br/>
                    <div>Your new home awaits at <Bold>555 Waverly 605 W 11th St, New York, Unit 3F</Bold> awaits.</div>
                    <br/>
                    <br/>
                    <ActionButton
                        onClick={() => history.push('/login')}>Click here</ActionButton>
                </div>
            </Fragment>

        );
    }
}

const mapStateToProps = state => ({
    leaseSettings: state.leaseSettings,
});

export default connect(mapStateToProps, null)(WelcomePage);
