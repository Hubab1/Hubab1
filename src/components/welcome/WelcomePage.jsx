import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { withTheme } from '@material-ui/core/styles';
import { Link } from 'react-router-dom';

import funnelImage from '../../assets/images/PoweredByFunnel.png';
import homeImage from '../../assets/images/home-image.png';
import Logo from 'components/common/Logo';
import { H1, H2, P }from 'assets/styles';
import {
    BackgroundImage, BackgroundImageTint, WelcomeFlexContainer, WelcomeTextContainer,
    WelcomeFooterContainer, HomeImageContainer
} from './styles';
import ActionButton from 'components/common/ActionButton/ActionButton';
import { ROUTES } from 'constants.js';


export class WelcomePage extends Component {

    getNextLinkUrl () {
        const hasClient = !!this.props.leaseSettings.client;
        const url = hasClient ? ROUTES.LOGIN : ROUTES.SIGNUP;
        return url;
    }
    
    render() {
        const { client, background_image, logo, name, street, city, state, postal_code, unit_number } = this.props.leaseSettings;
        const cityStateZip = `${city}, ${state} ${postal_code}`
        const helloContent = client ? `Hello ${client.first_name},` : 'Hi There,'

        return (
            <Fragment>
                <BackgroundImage url={background_image}/>
                <BackgroundImageTint primaryColor={this.props.theme.palette.primary.main}/>
                <WelcomeFlexContainer>
                    <Logo logo={logo}/>
                    <WelcomeTextContainer>
                        <HomeImageContainer>
                            <img src={homeImage} width="30" alt="company logo"/>
                        </HomeImageContainer>
                        <H2>
                            {helloContent}
                        </H2>
                        <P>Your new home awaits at</P>
                        {name && <H1>{name}</H1>}
                        {
                            name ? <P label="street">{street}</P> :
                            <H1>{street}</H1>
                        }
                        {cityStateZip && <P>{cityStateZip}</P>}
                        {unit_number && <P>{`Unit ${unit_number}`}</P>}
                    </WelcomeTextContainer>
                    <WelcomeFooterContainer>
                        <Link to={this.getNextLinkUrl()}>
                            <ActionButton
                                color="secondary"
                            >
                                Start Application
                            </ActionButton>
                        </Link>
                        <img src={funnelImage} width="200" style={{marginTop:'20px'}} alt="funnel logo" />
                    </WelcomeFooterContainer>
                </WelcomeFlexContainer>
            </Fragment>

        );
    }
}

const mapStateToProps = state => ({
    leaseSettings: state.leaseSettings,
});

const connectedWelcomePage = connect(mapStateToProps, null)(WelcomePage);
export default withTheme()(connectedWelcomePage);
