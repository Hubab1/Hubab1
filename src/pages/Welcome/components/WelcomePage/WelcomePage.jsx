import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { makeStyles, useTheme } from '@material-ui/core';
import clsx from 'clsx';

import { ROUTES } from 'constants/constants';
import ActionButton from 'components//ActionButton/ActionButton';
import { H1, H2, P, link } from 'assets/styles';
import funnelImage from 'assets/images/PoweredByFunnel.png';
import homeImage from 'assets/images/home-image.png';

const useStyles = makeStyles((theme) => {
    return {
        background: {
            position: 'absolute',
            top: 0,
            left: 0,
            width: '100%',
            height: '100%',
            backgroundPosition: 'center',
            backgroundSize: 'cover',
            zIndex: -5,
            opacity: theme.darkMode ? 1 : 0.3,
            backgroundImage: `url(${theme.assets.background})`,
        },
        backgroundOverlay: {
            position: 'absolute',
            top: 0,
            left: 0,
            width: '100%',
            height: '100%',
            zIndex: -4,
            opacity: 0.6,
            backgroundColor: theme.darkMode ? theme.palette.primary.main : 'none',
        },
        root: {
            position: 'absolute',
            display: 'flex',
            flexFlow: 'column',
            alignContent: 'space-around',
            color: 'white',
            height: '100%',
            width: '100%',
        },
        banner: {
            margin: '20px auto 0 auto',
            '& img': {
                maxWidth: '120px',
                maxHeight: '50px',
            },
        },
        content: {
            position: 'relative',
            margin: 'auto',
            paddingBottom: '50px',
            width: '82%',
            maxWidth: '500px',
            textAlign: 'center',
            backgroundColor: 'rgba(0, 0, 0, 0.6)',
        },
        homeImageContainer: {
            position: 'relative',
            top: '-35px',
            width: '70px',
            height: '35px',
            borderTopLeftRadius: '70px',
            borderTopRightRadius: '70px',
            background: 'inherit',
            textAlign: 'center',
            margin: '0 auto',

            '& img': {
                position: 'relative',
                top: '9px',
                width: '27px',
                height: '27px',
                objectFit: 'contain',
            },
        },
        footer: {
            display: 'flex',
            flexFlow: 'column',
            alignItems: 'center',
            margin: '2% 10%',
            height: '15%',
        },
        ctaButton: {
            backgroundColor: theme.darkMode ? theme.palette.common.white : theme.palette.primary.main,
        },
        link: {
            color: theme.darkMode ? theme.palette.common.white : theme.palette.common.black,
        },
    };
});

export function WelcomePage(props) {
    const theme = useTheme();
    const classes = useStyles();
    const { configuration } = props;
    const { community, unit } = configuration;
    const { building_name, city, state, postal_code, normalized_street_address } = community;

    const firstName = useMemo(() => {
        const { person, invitee } = configuration;
        if (person) {
            return person.first_name;
        } else if (invitee && invitee.first_name) {
            return invitee.first_name;
        } else {
            return null;
        }
    }, [configuration]);

    const cityStateZip = `${city}, ${state} ${postal_code}`;
    const helloContent = firstName ? `Hello ${firstName},` : 'Hi There,';

    return (
        <>
            <div className={classes.background} />
            <div className={classes.backgroundOverlay} />
            <div className={classes.root}>
                <div className={classes.banner}>
                    <img className={classes.logo} src={theme?.asserts?.logo} alt="company logo" />
                </div>
                <div data-testid="content" className={classes.content}>
                    <div className={classes.homeImageContainer}>
                        <img src={homeImage} alt="home-icon" />
                    </div>
                    <H2>{helloContent}</H2>
                    <P>Your new home awaits at</P>
                    {building_name && <H1 className="welcome__building-header">{building_name}</H1>}
                    {building_name ? (
                        <P>{normalized_street_address}</P>
                    ) : (
                        <H1 className="welcome__building-header">{normalized_street_address}</H1>
                    )}
                    {cityStateZip && <P>{cityStateZip}</P>}
                    {unit && unit.unit_number && <P>{`Unit ${unit.unit_number}`}</P>}
                </div>
                <div className={classes.footer}>
                    <Link to={{ pathname: ROUTES.SIGNUP }} style={{ textDecoration: 'none' }} className="cta-container">
                        <ActionButton data-testid="cta-button" buttonClassName={classes.ctaButton}>
                            Start Application
                        </ActionButton>
                    </Link>
                    <Link to={ROUTES.LOGIN} className={clsx(link, classes.link)}>
                        I already started an application
                    </Link>
                    <img src={funnelImage} width="150" alt="funnel logo" />
                </div>
            </div>
        </>
    );
}

WelcomePage.propTypes = {
    configuration: PropTypes.object.isRequired,
};

const mapStateToProps = (state) => ({
    configuration: state.configuration,
});

export default connect(mapStateToProps, null)(WelcomePage);
