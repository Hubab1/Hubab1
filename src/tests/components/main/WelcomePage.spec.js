import React from 'react';
import renderer from 'react-test-renderer';

import { WelcomePage } from 'components/main/WelcomePage';


it('matches snapshot for welcome page', function(){
    expect(
        renderer.create(<WelcomePage />)
    ).toMatchSnapshot();
});
