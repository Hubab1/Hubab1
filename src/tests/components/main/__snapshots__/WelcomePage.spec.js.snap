// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot for welcome page 1`] = `
Array [
  <div
    className="css-lwkisb"
  >
    <div>
      Jane Morgan
    </div>
    <span
      aria-label="agent-avatar"
      role="img"
      style={
        Object {
          "fontSize": 60,
          "lineHeight": "80px",
        }
      }
    >
      ðŸ‘©
    </span>
    <div
      className="css-pwky2g"
    >
      <div>
        Leasing Agent
      </div>
      <div>
        Sky Residencies
      </div>
    </div>
  </div>,
  <div
    style={
      Object {
        "padding": 25,
        "textAlign": "left",
      }
    }
  >
    <div
      className="css-vfetyo"
    >
      Hi Sam,
    </div>
    <div>
      Your magically simple rental application starts here.
    </div>
    <br />
    <div>
      Your new home awaits at 
      <span
        className="css-1xxe9l9"
      >
        555 Waverly 605 W 11th St, New York, Unit 3F
      </span>
       awaits.
    </div>
    <br />
    <br />
    <button
      className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedPrimary-13 MuiButton-raised-15 MuiButton-raisedPrimary-16 MuiButton-fullWidth-26"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex="0"
      type="submit"
    >
      <span
        className="MuiButton-label-2"
      >
        Click here
      </span>
      <span
        className="MuiTouchRipple-root-30"
      />
    </button>
  </div>,
]
`;
